### PR TITLE
Generate most common values for ColumnStats

### PR DESCRIPTION
**Pull request by mfussenegger**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql/src/main/java/io/crate/statistics/MostCommonValues.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+         * Note that we deliberately do this by removing values from the full
+         * list, rather than starting with an empty list and adding values,
+         * because the latter approach can fail to add any values if all the most
+         * common values have around the same frequency and make up the majority
+         * of the table, so that the overall average frequency of all values is
+         * roughly the same as that of the common values.  This would lead to any
+         * uncommon values being significantly overestimated.
+         */
+        int sumCount = 0;
+        for (int i = 0; i < numMcv - 1; i++) {
```
Comment (seut) [inline sql/src/main/java/io/crate/statistics/MostCommonValues.java]: `i < numMcv - 1` looks like the last item in `mcvCounts` is not accounted. Do I miss something here?

File: sql/src/test/java/io/crate/statistics/ColumnStatsTest.java [hunk 2]
```diff
@@ -39,6 +45,39 @@ public void test_column_stats_generation() {
...
+    @Test
+    public void test_column_stats_generation_for_num_samples_gt_mcv_target() {
+        List<Integer> numbers = IntStream.concat(
+            IntStream.generate(() -> 10).limit(90),
+            IntStream.concat(
+                IntStream.generate(() -> 20).limit(20),
+                IntStream.range(30, 150)
+            )
+        ).boxed()
+            .collect(Collectors.toList());
```
Comment (seut) [inline sql/src/test/java/io/crate/statistics/ColumnStatsTest.java]: formatting nitpick: indent looks wrong

File: sql/src/main/java/io/crate/statistics/MostCommonValues.java (hunk 1 above)
Comment (mfussenegger) [reply sql/src/main/java/io/crate/statistics/MostCommonValues.java]: This is intentionally, see the comment:

>  Here sumcount tracks the total count of all but the last (least common)
> * value in the MCV list, allowing us to determine the effect of excluding
> * that value from the list.

Should I rename the variable to `sumCountExclLast` or something like that?

Comment (seut) [reply sql/src/main/java/io/crate/statistics/MostCommonValues.java]: Ah right, sry missed that. It's fine for me as it is.

**Pull request merged**